File: test/unit/emotion_data_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:jordanian_ser_app/models/emotion.dart';

void main() {
  group('EmotionData Tests', () {
    test('All 4 core emotion categories are configured', () {
      expect(EmotionData.all.length, equals(4));
      expect(EmotionData.all.containsKey(EmotionType.angry), isTrue);
      expect(EmotionData.all.containsKey(EmotionType.happy), isTrue);
      expect(EmotionData.all.containsKey(EmotionType.sad), isTrue);
      expect(EmotionData.all.containsKey(EmotionType.neutral), isTrue);
    });

    test('Each emotion has at least 4 authentic prompts', () {
      for (final entry in EmotionData.all.entries) {
        final data = entry.value;
        expect(data.prompts.length, greaterThanOrEqualTo(4));
        expect(data.labelArabic.isNotEmpty, isTrue);
        expect(data.subtitleArabic.isNotEmpty, isTrue);
        expect(data.emoji.isNotEmpty, isTrue);
        expect(data.apiTag.isNotEmpty, isTrue);
      }
    });

    test('promptFor rotates and cycles within bounds', () {
      final happy = EmotionData.all[EmotionType.happy]!;
      final count = happy.prompts.length;
      expect(happy.promptFor(0), equals(happy.prompts[0]));
      expect(happy.promptFor(count), equals(happy.prompts[0]));
      expect(happy.promptFor(count + 1), equals(happy.prompts[1]));
    });
  });
}

File: test/unit/emotion_data_test.js
import { EmotionType, EmotionData } from '../models/emotion';

describe('EmotionData Tests', () => {
  test('All 4 core emotion categories are configured', () => {
    const keys = Object.keys(EmotionData.all);
    expect(keys.length).toBe(4);
    expect(EmotionData.all).toHaveProperty(EmotionType.angry);
    expect(EmotionData.all).toHaveProperty(EmotionType.happy);
    expect(EmotionData.all).toHaveProperty(EmotionType.sad);
    expect(EmotionData.all).toHaveProperty(EmotionType.neutral);
  });

  test('Each emotion has at least 4 authentic prompts', () => {
    Object.values(EmotionData.all).forEach((data) => {
      expect(data.prompts.length).toBeGreaterThanOrEqual(4);
      expect(data.labelArabic.length).toBeGreaterThan(0);
      expect(data.subtitleArabic.length).toBeGreaterThan(0);
      expect(data.emoji.length).toBeGreaterThan(0);
      expect(data.apiTag.length).toBeGreaterThan(0);
    });
  });

  test('promptFor rotates and cycles within bounds', () => {
    const happy = EmotionData.all[EmotionType.happy];
    const count = happy.prompts.length;
    expect(happy.promptFor(0)).toBe(happy.prompts[0]);
    expect(happy.promptFor(count)).toBe(happy.prompts[0]);
    expect(happy.promptFor(count + 1)).toBe(happy.prompts[1]);
  });
});